### PR TITLE
new end point to deactivate a common

### LINKED ISSUE #880: Implement delete commons - API endpoints
Add an endpoint on the backend to delete a common
for https://github.com/daostack/common-monorepo/issues/722 

We shouldn't delete it at all. Need to archive it

### LINKED ISSUE #1122: Create a "deactivate common" endpoint
We want to give users to delete their commons, but we don't want to actually delete from the database, so instead we'll use an already existing property in the common entity called "active".

Create an endpoint `/commons/decativate` that will set that property to false. (in the common by id)

### PR DESCRIPTION
**Pull request by noamso1**

### CODE REVIEW

**Comment by noamso1** [comment]
you can test this new endpoint like this:
curl -H 'authorization: {"uid":"DsFp7lzLWqcsoygIWnZs"}' http://localhost:5003/common-dev-dea4e/us-central1/commons/deactivate -d 'commonId=2bxGi5yAUJeQG4Aqivii'

**Comment by ariel-dao** [comment]
@noamso1 looks good, have you tested it?
and can you please add it to postman?

File: functions/src/common/index.ts [hunk 1]
```diff
@@ -72,6 +73,18 @@ router.post('/update', async (req, res, next) => (
     })
 ));
 
+router.post('/deactivate', async (req, res, next) => {
+  await responseExecutor(
+    async () => {
```
Comment (SobolevYuri) [inline functions/src/common/index.ts:78]: why don't you verify the req.body by a schema?

File: functions/src/common/business/deactivateCommon.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+//curl -H 'authorization: {"uid":"DsFp7lzLWqcsoygIWnZs"}' http://localhost:5003/common-dev-dea4e/us-central1/commons/deactivate -d 'commonId=2bxGi5yAUJeQG4Aqivii'
+
+export const deactivateCommon = async (commonId: string, userId: string): Promise<IUpdatableCommon> => {
+
+  if ( ! await hasPermission(userId, commonId) ) {
+    throw new UnauthorizedError();
+  }
+
+  let updatedCommon = await commonDb.update(
```
Comment (SobolevYuri) [inline functions/src/common/business/deactivateCommon.ts]: const

**Comment by noamso1** [comment]
Yes. I tested it.
I don't have access to the postman team, can you send it to me?
btw I've added a curl payload example to the code at
functions/src/common/business/deactivateCommon.ts
line 15

**Review by SobolevYuri — CHANGES_REQUESTED** [review]
check the comments, pls

**Comment by noamso1** [comment]
@SobolevYuri  I did all your recommendations! please check.

**Review by SobolevYuri — APPROVED** [review]
great! thanks =)

**Pull request merged**